src/main/java/com/exasol/adapter/document/files/FileTypeSpecificDocumentFetcherFactoryInterface.java

```diff
@@ -0,0 +1,28 @@
+package com.exasol.adapter.document.files;
+
+import java.util.List;
+
+import com.exasol.adapter.document.documentfetcher.DocumentFetcher;
+
+/**
+ * Interface for factories for {@link FileTypeSpecificDocumentFetcher}.
+ */
+public interface FileTypeSpecificDocumentFetcherFactoryInterface {
```

Comment:
```suggestion
public interface FileTypeSpecificDocumentFetcherFactory {
```

Reply:
I used the interface ending since `FileTypeSpecificDocumentFetcherFactory` is already taken by the dispatching factory.